controller/src/beerocks/cli/beerocks_cli_socket.h

```diff
@@ -21,11 +21,11 @@ class cli_socket : public socket_thread, public cli {
     cli_socket(std::string temp_path_, std::string proxy_ip_ = std::string());
     ~cli_socket();
 
-    bool connect();
-    void disconnect();
-    bool is_connected() { return (master_socket != nullptr); }
-    bool start() { return socket_thread::start(); }
-    void stop() { socket_thread::stop(); }
+    bool connect() override;
```

Comment:
```suggestion
    virtual bool connect() override;
```
Its not mandatory and is exactly the same functionality-wise, but is clearer.